Islandora-CLAW/islandora_demo
# Islandora Demo Default Content Model Update

**GitHub Issue**: [#896](https://github.com/Islandora-CLAW/CLAW/issues/896)

# What does this Pull Request do?

Adds several fields to the default content model.

# What's new?

- New islandora_object fields:
    - field_alternative_title
    - field_description
    - field_edtf_date
    - field_edtf_date_created
    - field_edtf_date_issued
    - field_extent
    - field_identifier
    - field_linked_agent (uses the new RelationType field from controlled_access_terms and populated with MARC relators; resolves [CLAW/issues/893](https://github.com/Islandora-CLAW/CLAW/issues/893))
    - field_member_of
    - field_resource_type
    - field_rights
    - field_subject
- updated rdf.mapping
- new vocabulary for resource types (partially addresses [CLAW/issues/894](https://github.com/Islandora-CLAW/CLAW/issues/894))
- adds dependencies on controlled_access_terms_default_configuration
- implements hook_rdf_namespaces in islandora_demo.module 

# How should this be tested?

From a fresh and current claw-playbook build:

- enable controlled_access_terms_default_configuration
- Apply the PR
- using features, import the changes to the islandora_demo feature
- Make a new repository object (everything should be there and all the fields working properly).

# Additional Notes:

The JSON-LD still has a few quirks with the new profile:
- the new date fields will be given a string type. (See https://github.com/Islandora-CLAW/CLAW/issues/889)
- linked agents will use the contributor predicate regardless of the drop-down. (Related to https://github.com/Islandora-CLAW/CLAW/issues/889)

# Interested parties
@Islandora-CLAW/committers @dannylamb @rosiel
